xbmc/utils/SysfsUtils.cpp

```diff
@@ -7,114 +7,85 @@
  */
 
 #include "SysfsUtils.h"
+
 #include "utils/log.h"
-#include "utils/StringUtils.h"
 
-#include <unistd.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <fcntl.h>
-#include <string.h>
+#include <fstream>
 
-#ifdef TARGET_WINDOWS_STORE
-#include <io.h>
-#endif
+#include <unistd.h>
 
-int SysfsUtils::SetString(const std::string& path, const std::string& valstr)
+namespace
 {
-  int fd = open(path.c_str(), O_RDWR, 0644);
-  int ret = 0;
-  if (fd >= 0)
-  {
-    if (write(fd, valstr.c_str(), valstr.size()) < 0)
-      ret = -1;
-    close(fd);
-  }
-  if (ret)
-    CLog::Log(LOGERROR, "%s: error writing %s",__FUNCTION__, path.c_str());
-
-  return ret;
+const int PAGESIZE = getpagesize();
 }
 
-int SysfsUtils::GetString(const std::string& path, std::string& valstr)
+template<>
+std::string CSysfs::Get()
 {
-  int len;
-  char buf[256] = {0};
+  std::ifstream file(m_path);
 
-  int fd = open(path.c_str(), O_RDONLY);
-  if (fd >= 0)
-  {
-    valstr.clear();
-    while ((len = read(fd, buf, 256)) > 0)
-      valstr.append(buf, len);
-    close(fd);
+  if (!file.is_open())
+    return "";
 
-    StringUtils::Trim(valstr);
```

Comment:
Related to @MilhouseVH comment: `Trim()` is gone now

Reply:
I'll address this in another PR